Sources/StreamChat/ChatClient.swift

```diff
@@ -404,6 +405,48 @@ public class ChatClient {
             completion()
         }
     }
+    
+    // MARK: - Listening for Client Events
+    
+    /// Subscribes to web-socket events of the specified event type.
+    ///
+    /// - Note: The handler is always called on the main thread.
+    ///
+    /// An example of observing connection status changes:
+    /// ```swift
+    /// client.subscribe(to: ConnectionStatusUpdated.self) { connectionEvent in
+    ///     switch connectionEvent.connectionStatus {
+    ///         case .connected:
+    ///           …
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// - SeeAlso: ``Chat.subscribe(to:handler:)`` for subscribing to channel specific events.
```

Comment:
TBH I don't think we need this 🤔 No other SDK has a component to observer only Channel events. The new `ChatClient.subscribe(toEvent:)` is more than enough IMO

Reply:
We had a longer discussion on this in the tech meeting and the end result was that let's keep it as is and revisit it when we can use the new enum based events.